lib/extended-css-selector.js

```diff
@@ -210,11 +210,21 @@ var ExtendedSelector = (function () { // jshint ignore:line
             var re = /\[-ext-([a-z-_]+)=(["'])((?:(?=(\\?))\4.)*?)\2\]/g;
             var str = selectorText.replace(re, evaluateMatch);
 
+            // Sizzle's parsing of pseudo class arguments is buggy on certain circumstances
+            // We support following form of arguments:
+            // 1. for :matches-css, those of a form {propertyName}: /.*/
+            // 2. for :contains, those of a form /.*/
+            // We transform such cases in a way that Sizzle has no ambiguity in parsing arguments.
+            str = str.replace(/\:(matches-css(?:-after|-before)?)\(([a-z-\s]*\:\s*\/(?:\\.|[^\/])*?\/\s*)\)/g, addQuotes);
+            str = str.replace(/:(contains|has-text)\((\s*\/(?:\\.|[^\/])*?\/\s*)\)/g, addQuotes);
+            // Note that we require `/` character in regular expressions to be escaped.
+
             var compiledSelector = tokenizeSelector(str);
 
             // Compiles and validates selector
             // Compilation in Sizzle means that selector will be saved to the inner cache and then reused
-            Sizzle.compile(selectorText);
+            // Directly compiling unprocessed selectorText can cause problems. Only those which are already processed should be compiled.
```

Comment:
It makes sense to mention that test in the comment just for the reader to better understand